File: Daily-DSA/Day-26/C++/LinearSearch.cpp

// Find the Unique Element

#include <iostream>
#include <vector>
using namespace std;

class FindUnique {
    public:
        int xor_tool = 0;

        void unique(vector<int> array) {
            for(int val : array){
                xor_tool ^= val;
            }
            cout << "The Unique element is :- " << xor_tool << "\n";
        }
};

int main () {
    vector<int> array = {2, 3, 6, 5, 3, 7, 6, 7, 2};

    FindUnique Element;
    Element.unique(array);

    return 0;
}

File: Daily-DSA/Day-26/Golang/LinearSearch.go
// Find the Unique Element

package main

import "fmt"

func main() {
	numbers := [9]int{2, 3, 6, 5, 3, 7, 6, 7, 2}
	var xor_tool int = 0

	for _, value := range numbers {
		xor_tool ^= value
	}

	fmt.Printf("The Unique element is :- %d \n", xor_tool)
}
